genie-web/src/main/java/com/netflix/genie/web/configs/GenieAgentFilterAutoConfiguration.java

```diff
@@ -0,0 +1,131 @@
+/*
+ *
+ *  Copyright 2016 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.configs;
+
+import com.netflix.genie.web.properties.AgentFilterProperties;
+import com.netflix.genie.web.services.AgentFilterService;
+import com.netflix.genie.web.services.impl.AgentFilterServiceImpl;
+import com.netflix.genie.web.util.AgentMetadataInspector;
+import com.netflix.genie.web.util.BlacklistedVersionAgentMetadataInspector;
+import com.netflix.genie.web.util.InspectionReport;
+import com.netflix.genie.web.util.MinimumVersionAgentMetadataInspector;
+import com.netflix.genie.web.util.WhitelistedVersionAgentMetadataInspector;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.annotation.Order;
+
+import java.util.List;
+
+/**
+ * Auto-configuration for the {@link AgentFilterService} default implementation.
+ * This component is activated unless another bean is present AND if the corresponding property is set to true.
+ * <p>
+ * This configuration also creates a set of pre-configured {@link AgentMetadataInspector} that the
+ * service loads and uses:
+ * - Version whitelist: accept agent IFF version matches a given pattern
+ * - Version blacklist: reject agent whose version matches a given pattern
+ * - Minimum version: reject agent whose version is lower than a given minimum
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Configuration
+@ConditionalOnProperty(value = AgentFilterProperties.VERSION_FILTER_ENABLED_PROPERTY, havingValue = "true")
```

Comment:
I'm not sure but verify the default setting for `match if missing` is what you want or explicitly set it

Reply:
`matchIfMissing` is default `false` which is what I want.